code/modules/food_and_drinks/food.dm

```diff
@@ -1,12 +1,156 @@
+	var/ticksRotted = 0 // how much we've rotted, seperate from just checkign toxins for "poison" food, like carp
+	var/visibleRot = FALSE // are we now visibly rotting
+
+
 /obj/item/weapon/reagent_containers/food/New()
-		..()
-		pixel_x = rand(-5, 5)	//Randomizes postion slightly.
-		pixel_y = rand(-5, 5)
\ No newline at end of file
+	..()
+	pixel_x = rand(-5, 5)	//Randomizes postion slightly.
+	pixel_y = rand(-5, 5)
+	SSfood.insertFood(src)
+	if(targetTemperature >= 50) // gotcha to make sure any missed flags dont break
+		shouldBeHot = TRUE
+	if(!src.reagents)
+		canRot = FALSE // force off rotting for any non-reagent foods
+
+/obj/item/weapon/reagent_containers/food/initialize()
+
+// degrade handling
+/obj/item/weapon/reagent_containers/food/proc/updateFood()
+	if(src.loc) // less intensive but hackier way to make sure frozen food is fine
+		if(is_type_in_typecache(src.loc, whitelist_foodlocs))
+			reachTemp = -50
+	if(canRot)
+		if(!src.reagents)
+			create_reagents(DEFAULT_REAGENT_SIZE)
+			src.reagents.add_reagent("nutriment",1) // if we somehow got here with not having a reagent holder, make one and give us a tiny bit of nutriment to work with.
+		if(lifetime > 0)
+			--lifetime
+		var/turf/T = get_turf(src)
+		if(T)
+			if(reachTemp < T.temperature)
+				reachTemp += 0.1
+			else if (reachTemp > T.temperature)
+				reachTemp -= 0.1
+
+			if(temperature < reachTemp)
+				temperature += 1
+			else if (temperature > reachTemp)
+				temperature -= 1
+		else
+			reachTemp = targetTemperature // if we're in nullspace or just not able to get our pos, preserve our selves
+
+		if((temperature < (targetTemperature - 1) && shouldBeHot) || (temperature > (targetTemperature + 1) && !shouldBeHot))
+			freshness += (lifetime > 0 ? 0 : degradeSpeed) // we're rotting, panic!
+		else if((temperature < (targetTemperature - 1) && !shouldBeHot) || (temperature > (targetTemperature + 1) && shouldBeHot))
+			freshness -= (lifetime > 0 ? 0 : degradeSpeed) // we're getting a bit better.
+
+		if(meltable && temperature > meltingPoint) // we jelly now
+			var/obj/effect/decal/cleanable/molten_item/MI = new/obj/effect/decal/cleanable/molten_item(get_turf(src))
+			if(src.reagents)
+				src.reagents.trans_to(MI, src.reagents.total_volume, 1, 1, 1)
+			qdel(src)
+
+		freshness = Clamp(freshness,-100,100)
+
+		if(freshness > 0)
+			for(var/datum/reagent/R in src.reagents)
+				for(var/datum/reagent/check in bestReagents)
+					if(R.id == check.id && R.volume >= check.volume) // no magical creation of reagents, only adjust whats still in us and nothing else
+						R.volume = check.volume * (freshness / 100)
+		else if(freshness < 0) // we're beginning to rot
+			if(freshness < mouldRating)
+				ticksRotted++
+				src.reagents.add_reagent("toxin",1)
+				if(ticksRotted > 50 && !visibleRot)
+					visibleRot = TRUE
+					var/image/stank = image('icons/effects/effects.dmi', icon_state = "stinky")
```

Comment:
@Jalleo  Perhaps, but it's a one time thing. Food ain't going to un-rot, and it's not going to be referenced at all, so I thought it best to just do it easily like that.